src/main/kotlin/io/meshcloud/dockerosb/service/GenericServiceInstanceBindingService.kt

```diff
@@ -18,9 +18,11 @@ import java.util.*
 class GenericServiceInstanceBindingService(
     private val yamlHandler: YamlHandler,
     private val gitHandler: GitHandler,
-    private val catalog: Catalog
+    catalogService: GenericCatalogService
 ) : ServiceInstanceBindingService {
 
+    private val catalog = catalogService.getCatalogInternal()
```

Comment:
f: If you do it like this, this reference to the catalog will only be set when this bean is created. Any subsequent updates to the catalog will not be reflected in the variable. Did you test this manually to verify?